src/jit/morph.cpp

```diff
@@ -3259,43 +3111,13 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
         bool                 isNonStandard = false;
         regNumber            nonStdRegNum  = REG_NA;
 
-        fgArgTabEntry* argEntry = nullptr;
-
-        if (reMorphing)
-        {
-            argEntry = gtArgEntryByArgNum(call, argIndex);
-        }
-
-        // Setup any HFA information about 'argx'
-        var_types hfaType  = TYP_UNDEF;
-        bool      isHfaArg = false;
-        unsigned  hfaSlots = 0;
-
 #ifdef FEATURE_HFA
-        if (reMorphing)
+        hfaType = GetHfaType(argx);
+        if (varTypeIsFloating(hfaType) && !callIsVararg)
```

Comment:
`if (varTypeIsFloating(hfaType) && !callIsVararg)`

This is not strictly true, this is target dependent. Only on Windows arm/arm64 this is a true statement. The existing code had this under an ifdef correctly. Specifically, unix arm/arm64 will still need to classify HFA types for native varargs methods.

Reply:
@jashook - If I understand correctly, even arm64 has to always classify hfa types, because they are passed by value even for varargs, though by size they might be too large. The existing code was separately tracking that. Are you saying that the setting of `compFloatingPointUsed` needs to be done on non-windows varargs? 
Note that this code unconditionally gets the HFA type, and then the `isHfaArg` boolean tracks whether it's actually passed in floating point registers.
Could you look at the comment near line 3650 in the new code and see if it is correct?

Reply:
```   // On ARM64, HFA types that might otherwise be passed by reference due to their size
        // will still be passed by value, so we need to record their hfaType.
        // On ARM, we don't treat HFA types differently if they are not passed in registers,
        // but in both cases we still keep track of whether we have an HFA.
```
Is correct for everything but arm64 windows varargs. The HFA type will be dropped/ignored for varargs arm64 Windows we will classify as if they were not HFAs. They will instead follow the calling convention for passing an non-HFA struct, and will force the floating point values to be passed in integer registers. However, unix arm64 does not have this restriction, and will treat a vararg call with no difference from the regular HFA classification, therefore pass in floating point registers.